src/Control/Eff/Cut.hs

```diff
@@ -2,6 +2,8 @@
 {-# LANGUAGE PatternGuards #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
```

Comment:
ok so this seems related to [trac #8883](https://ghc.haskell.org/trac/ghc/ticket/8883) and [trac #9558](https://ghc.haskell.org/trac/ghc/ticket/8883) and is [an intended change after ghc-7.8](https://downloads.haskell.org/~ghc/7.10.1-rc1/docs/html/users_guide/release-7-10-1.html) (i.e., disallowing ghc to infer types you cannot otherwise write given the enabled language pragmas).

your fix seems like the reasonable one, approving